### PR TITLE
[build][test][tensorflow] Add TF to TF 1.15 inference DLCs

### PR DESCRIPTION
**Pull request by saimidu**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
Install TensorFlow on TF 1.15 inference DLCs

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [x] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [x] `sagemaker_local_tests = true`

### DLC image/dockerfile
- tensorflow/inference/docker/1.15/py36/Dockerfile.cpu
- tensorflow/inference/docker/1.15/py36/cu100/Dockerfile.gpu

## Label Checklist
- [x] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [x] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [x] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [x] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [x] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: dlc_developer_config.toml [hunk 1]
```diff
@@ -19,7 +19,7 @@ benchmark_mode = false
 
 [build]
 # Frameworks for which you want to disable both builds and tests
-skip_frameworks = []
+skip_frameworks = ["mxnet", "pytorch", "autogluon", "huggingface_tensorflow", "huggingface_pytorch"]
```
Comment (saimidu) [inline dlc_developer_config.toml:22]: To be reverted.

File: src/main.py [hunk 2]
```diff
@@ -44,8 +44,8 @@ def main():
     # TODO: Remove when we remove these jobs completely
     build_arn = utils.get_codebuild_build_arn()
     if build_context == "PR":
-        tf_1_build_regex = re.compile(r"dlc-pr-tensorflow-1:")
-        if tf_1_build_regex.search(build_arn):
+        tf_2_build_regex = re.compile(r"dlc-pr-tensorflow-2:")
+        if tf_2_build_regex.search(build_arn):
```
Comment (saimidu) [inline src/main.py:48]: To be reverted.

File: tensorflow/buildspec-tf1.yml [hunk 3]
```diff
@@ -39,114 +39,6 @@ context:
       target: deep_learning_container.py
 
 images:
-  BuildTensorflowCPUTrainPy36DockerImage:
```
Comment (saimidu) [inline tensorflow/buildspec-tf1.yml:42]: To be reverted.

File: test/dlc_tests/sanity/test_safety_report_file.py [hunk 4]
```diff
@@ -65,7 +65,7 @@ def test_safety_file_exists_and_is_valid(image):
     :param image: str, image uri
     """
     repo_name, image_tag = image.split("/")[-1].split(":")
-    container_name = f"{repo_name}-{image_tag}-safety"
+    container_name = f"{repo_name}-{image_tag}-safety-file"
```
Comment (saimidu) [inline test/dlc_tests/sanity/test_safety_report_file.py:68]: Unrelated test fix.

File: test/requirements.txt [hunk 5]
```diff
@@ -1,3 +1,4 @@
+dataclasses
```
Comment (saimidu) [inline test/requirements.txt:1]: Unrelated test fix.

**Comment by SergTogul** [comment]
LGTM

File: test/sagemaker_tests/tensorflow/inference/test/integration/sagemaker/conftest.py [hunk 6]
```diff
@@ -256,18 +256,40 @@ def disable_test(request):
...
+#     But the flag "--last-failed-no-failures all" requires pytest to execute all the available tests.
+#     The only sign that a test passed last time - lastfailed file exists and the test name isn't in that file.
+#     The method checks whether lastfailed file exists and the test name is not in it.
+#     """
+#     test_name = request.node.name
+#     lastfailed = request.config.cache.get("cache/lastfailed", None)
+#
+#     if lastfailed is not None \
+#             and not any(test_name in failed_test_name for failed_test_name in lastfailed.keys()):
+#         pytest.skip(f"Skipping {test_name} because it was successfully executed for this commit")
```
Comment (saimidu) [inline test/sagemaker_tests/tensorflow/inference/test/integration/sagemaker/conftest.py:295]: Will revert this commented fixture code once PR is approved.

**Review by SergTogul — DISMISSED** [review]

File: test/sagemaker_tests/tensorflow/inference/test/integration/sagemaker/test_tfs.py [hunk 7]
```diff
@@ -160,3 +168,43 @@ def test_python_model_with_lib(boto_session, sagemaker_client,
     # python service adds this to tfs response
     assert output_data["python"] is True
     assert output_data["dummy_module"] == "0.1"
+
+
+@pytest.mark.model("resnet50")
+def test_resnet_with_inference_handler(
```
Comment (arjkesh) [inline test/sagemaker_tests/tensorflow/inference/test/integration/sagemaker/test_tfs.py:174]: did you test this on TF2.5?

File: test/sagemaker_tests/tensorflow/inference/test/integration/sagemaker/timeout.py [hunk 8]
```diff
@@ -0,0 +1,83 @@
+# Copyright 2019-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (arjkesh) [inline test/sagemaker_tests/tensorflow/inference/test/integration/sagemaker/timeout.py:1]: Is this copied from somewhere? copyright looks old

File: test/sagemaker_tests/tensorflow/inference/test/integration/sagemaker/timeout.py [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+def timeout_and_delete_endpoint(endpoint_name, sagemaker_session,
+                                seconds=0, minutes=0, hours=0):
+    with timeout(seconds=seconds, minutes=minutes, hours=hours) as t:
+        try:
+            yield [t]
+        finally:
+            try:
+                sagemaker_session.delete_endpoint(endpoint_name)
+                LOGGER.info("deleted endpoint {}".format(endpoint_name))
+            except ClientError as ce:
```
Comment (arjkesh) [inline test/sagemaker_tests/tensorflow/inference/test/integration/sagemaker/timeout.py:65]: I believe these will suppress all client errors if you don't have a condition to raise. You may want to at least log information if the endpoint fails to delete

File: test/sagemaker_tests/tensorflow/inference/test/integration/sagemaker/timeout.py [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+@contextmanager
+def timeout_and_delete_endpoint_by_name(endpoint_name, sagemaker_session, seconds=0, minutes=0, hours=0):
+    with timeout(seconds=seconds, minutes=minutes, hours=hours) as t:
+        try:
+            yield [t]
+        finally:
+            try:
+                sagemaker_session.delete_endpoint(endpoint_name)
+                LOGGER.info('deleted endpoint {}'.format(endpoint_name))
+            except ClientError as ce:
```
Comment (arjkesh) [inline test/sagemaker_tests/tensorflow/inference/test/integration/sagemaker/timeout.py:80]: I believe these will suppress all client errors if you don't have a condition to raise. You may want to at least log information if the endpoint fails to delete

File: test/sagemaker_tests/tensorflow/inference/test/integration/sagemaker/timeout.py [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+
+from botocore.exceptions import ClientError
+
+LOGGER = logging.getLogger('timeout')
+
+
+class TimeoutError(Exception):
+    pass
+
+
```
Comment (arjkesh) [inline test/sagemaker_tests/tensorflow/inference/test/integration/sagemaker/timeout.py:27]: nit: using .format instead of f strings in this file

File: test/sagemaker_tests/tensorflow/inference/test/integration/sagemaker/conftest.py [hunk 12]
```diff
@@ -256,18 +256,40 @@ def disable_test(request):
...
-    The method checks whether lastfailed file exists and the test name is not in it.
-    """
-    test_name = request.node.name
-    lastfailed = request.config.cache.get("cache/lastfailed", None)
-
-    if lastfailed is not None \
-            and not any(test_name in failed_test_name for failed_test_name in lastfailed.keys()):
-        pytest.skip(f"Skipping {test_name} because it was successfully executed for this commit")
+@pytest.fixture(scope="session")
+def resnet_model_tar_path():
```
Comment (arjkesh) [inline test/sagemaker_tests/tensorflow/inference/test/integration/sagemaker/conftest.py:260]: Can you add a docstring here? There seems to be some pretty specific path/s3 downloading logic, would be good to document why it's needed

File: test/sagemaker_tests/tensorflow/inference/test/integration/sagemaker/timeout.py (hunk 8 above)
Comment (saimidu) [reply test/sagemaker_tests/tensorflow/inference/test/integration/sagemaker/timeout.py:1]: I copied it from the PT inference tests. Let me update the text.

File: test/sagemaker_tests/tensorflow/inference/test/integration/sagemaker/test_tfs.py (hunk 7 above)
Comment (saimidu) [reply test/sagemaker_tests/tensorflow/inference/test/integration/sagemaker/test_tfs.py:174]: Good point. I tried it only once, but I didn't really use this same script. So I'll retry it to ensure that it works correctly.

File: test/sagemaker_tests/tensorflow/inference/test/integration/sagemaker/timeout.py (hunk 9 above)
Comment (saimidu) [reply test/sagemaker_tests/tensorflow/inference/test/integration/sagemaker/timeout.py:65]: I haven't given much thought about how to do this, or why we currently suppress all errors. I'd like us to dedicate time to fixing this on both TF and PT inference, later, though I will add a `raise` to this exception handler.

File: test/sagemaker_tests/tensorflow/inference/test/integration/sagemaker/timeout.py (hunk 11 above)
Comment (saimidu) [reply test/sagemaker_tests/tensorflow/inference/test/integration/sagemaker/timeout.py:27]: Will update.

File: test/sagemaker_tests/tensorflow/inference/test/integration/sagemaker/conftest.py (hunk 12 above)
Comment (saimidu) [reply test/sagemaker_tests/tensorflow/inference/test/integration/sagemaker/conftest.py:260]: Sure, will do.
